sqlglot/optimizer/qualify_columns.py

```diff
@@ -441,46 +438,22 @@ def _expand_struct_stars(
             # There is no matching field in the struct
             return []
 
-    # Each nested field caches its path in exp.Identifier parts,
-    # e.g. tbl.f0.f1.item -> [exp.Identifier(tbl), ..., exp.Identifier(item)]
-    field_path: t.Dict[int, t.List[exp.Identifier]] = {id(starting_struct): []}
-
-    # The keys to non-struct field entries in `nested_fields`
-    field_keys = []
-
-    stack = [(starting_struct, id(starting_struct))]
+    name_set = set()
+    new_selections = []
 
-    # Traverse & flatten the starting struct
-    while stack:
-        field, parent_struct_id = stack.pop()
+    for field in t.cast(exp.DataType, starting_struct.kind).expressions:
+        name = field.name
 
-        # Unable to expand star unless all struct fields are named and annotated
-        if not isinstance(field.this, exp.Identifier) or not field.kind:
+        # Ambiguous or anonymous fields can't be expanded
+        if name in name_set or not isinstance(field.this, exp.Identifier):
             return []
 
-        field_id = id(field)
-
-        # Preserve the qualified path for this field
-        field_path[field_id] = [
-            part.copy() for part in itertools.chain(field_path[parent_struct_id], [field.this])
-        ]
+        name_set.add(name)
 
-        if not field.kind.is_type(exp.DataType.Type.STRUCT):
-            # Collect the nested non-struct fields
-            field_keys.append(field_id)
-        else:
-            stack.extend((expr, field_id) for expr in reversed(field.kind.expressions))
-
-    new_selections = []
-    for field_key in field_keys:
-        # The fully qualified path of each field is created by:
-        # - The path until the starting struct, if any (dot_parts)
-        # - The path from the starting struct until the nested field (field_path[field_key])
-        root, *parts = [
-            part.copy() for part in itertools.chain(dot_parts[:-1], field_path[field_key])
-        ]
+        this = field.this.copy()
+        root, *parts = [part.copy() for part in itertools.chain(dot_parts, [this])]
         new_column = exp.column(t.cast(exp.Identifier, root), table=dot_column.table, fields=parts)
```

Comment:
I think if the `dot_column`'s table contained quotes, you'd drop them in `table=dot_column.table` because under the hood it simply invokes `text("table")`. You need to do `dot_column.args.get("table")`.

Let's 1) ensure this _does_ indeed drop the quotes in an example where the table name is quoted, 2) fix it and 3) add a test to avoid similar regressions.

Please also check for similar property method invocations to ensure we don't silently drop quotes.